db/view/view.cc

```diff
@@ -2549,38 +2549,45 @@ void view_builder::on_drop_view(const sstring& ks_name, const sstring& view_name
 }
 
 future<> view_builder::do_build_step() {
-    return seastar::async([this] {
-        exponential_backoff_retry r(1s, 1min);
-        while (!_base_to_build_step.empty() && !_as.abort_requested()) {
-            auto units = get_units(_sem, 1).get();
-            ++_stats.steps_performed;
-            try {
-                execute(_current_step->second, exponential_backoff_retry(1s, 1min));
-                r.reset();
-            } catch (const abort_requested_exception&) {
-                return;
-            } catch (...) {
-                ++_current_step->second.base->cf_stats()->view_building_paused;
-                ++_stats.steps_failed;
-                auto base = _current_step->second.base->schema();
-                vlogger.warn("Error executing build step for base {}.{}: {}", base->ks_name(), base->cf_name(), std::current_exception());
-                r.retry(_as).get();
-                initialize_reader_at_current_token(_current_step->second).get();
-            }
-            if (_current_step->second.build_status.empty()) {
-                auto base = _current_step->second.base->schema();
-                auto reader = std::move(_current_step->second.reader);
-                _current_step = _base_to_build_step.erase(_current_step);
-                reader.close().get();
-            } else {
-                ++_current_step;
-            }
-            if (_current_step == _base_to_build_step.end()) {
-                _current_step = _base_to_build_step.begin();
+    // View building is initiated during a schema merge where the new view is added.
+    // The schema merge is performed in the gossip scheduling group, but the actual
+    // work of building the view, which may be time-consuming, is done in the streaming
+    // (maintenance) scheduling group, so that it doesn't impact the processing of tasks
+    // in the gossip scheduling group.
+    return with_scheduling_group(_db.get_streaming_scheduling_group(), [this] {
+        return seastar::async([this] {
```

Comment:
You can use `seastar::thread_attributes` to specify the scheduling group that the `seastar::async` will run in. If you do this, you can avoid indenting the body of the function.

Reply:
I was actually hoping something like this exists but didn't find it. I used this in the rebase